### PR TITLE
Add auto exit config types & mutations

### PR DESCRIPTION
**Pull request by ZachCMP**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add mutations & logic for updating Auto Exit Configs in the frontend

Relies on backend PR: https://github.com/greenriver/hmis-warehouse/pull/3820
Relies on frontend PR: https://github.com/greenriver/hmis-frontend/pull/581

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/graphql/mutations/delete_auto_exit_config.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+module Mutations
+  class DeleteAutoExitConfig < CleanBaseMutation
+    argument :id, ID, required: true
+
+    field :auto_exit_config, Types::HmisSchema::AutoExitConfig, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(id:)
+      record = Hmis::AutoExitConfig.find(id)
+      return { errors: [HmisErrors::Error.new(:auto_exit_config, :not_allowed)] } unless allowed?(permissions: [:can_configure_data_collection])
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/delete_auto_exit_config.rb]:
```suggestion
      raise 'Access denied' unless allowed?(permissions: [:can_configure_data_collection])
```

We've switched to using exceptions for unexpected failures so we can track them in sentry. It looks like the `default_create_record` is still returning a validation which is maybe where you copied this from . You can update that to an exception too.

File: drivers/hmis/app/graphql/mutations/update_auto_exit_config.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  class UpdateAutoExitConfig < CleanBaseMutation
+    argument :input, Types::HmisSchema::AutoExitConfigInput, required: true
+    argument :id, ID, required: true
+
+    field :auto_exit_config, Types::HmisSchema::AutoExitConfig, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(id:, input:)
+      record = Hmis::AutoExitConfig.find(id)
+      return { errors: [HmisErrors::Error.new(:auto_exit_config, :not_allowed)] } unless allowed?(permissions: [:can_configure_data_collection])
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/update_auto_exit_config.rb]:
```suggestion
      raise 'Access denied' unless allowed?(permissions: [:can_configure_data_collection])
```

File: drivers/hmis/app/models/hmis/auto_exit_config.rb [hunk 3]
```diff
@@ -7,6 +7,11 @@
 class Hmis::AutoExitConfig < Hmis::HmisBase
   self.table_name = 'hmis_auto_exit_configs'
 
+  belongs_to :project, optional: true, class_name: 'Hmis::Hud::Project'
+  belongs_to :organization, optional: true, class_name: 'Hmis::Hud::Organization'
+
+  validates :length_of_absence_days, numericality: { greater_than_or_equal_to: 30 }
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/auto_exit_config.rb:13]: I think this is good, it's causing a test failure though

**Comment by ZachCMP** [comment]
@gigxz Should be all fixed now. Implemented all your feedback

**Review by gigxz — APPROVED** [review]
looks good to me!

**Pull request merged**